src/nccl_ofi_net.c

```diff
@@ -2758,7 +2762,8 @@ static ncclResult_t ofi_iflush(void* recvComm, void* buffer, int size,
 	void *data = NULL;
 	void *flush_mr_desc = NULL;
 
-	if (ofi_nccl_gdr_flush_disable() || !support_gdr)
+	/* If flushing is disabled or not needed, just return. */
+	if (!gdr_flush || !support_gdr || (!local_mr && !hmem_mr))
```

Comment:
Why do we need to add `!local_mr && !hmem_mr`? Flush would only be used if underlying network supports transfers using GPU buffers. After which users can simply disable flush operations if they still wouldn't like to use it.